cmd/node_problem_detector.go

```diff
@@ -50,7 +81,9 @@ func main() {
 		os.Exit(0)
 	}
 
+	nodeName := mustGetNodeName()
+
 	k := kernelmonitor.NewKernelMonitorOrDie(*kernelMonitorConfigPath)
-	p := problemdetector.NewProblemDetector(k, *apiServerOverride)
+	p := problemdetector.NewProblemDetector(k, *apiServerOverride, nodeName)
```

Comment:
@Random-Liu I agree with you.

BTW, as we may add more command line argument, we may need to add a `Conf` struct and make all command line arguments as its fields. When we need a command line argument, we just need to reference an `Conf` instance. This will also make the code more readable. :)

I suggest we make another PR to make this refactor. WDYT.